registry/client/repository.go

```diff
@@ -208,8 +208,21 @@ type tags struct {
 // All returns all tags
 func (t *tags) All(ctx context.Context) ([]string, error) {
 	var tags []string
-
-	u, err := t.ub.BuildTagsListURL(t.name)
+	var u string
+	var err error
+	var compatUrl bool
+
+	// This is needed to be compatible with old registries that use "list" instead of "_list"
+	_, compatUrl = ctx.Value("tags.url.compat").(bool)
```

Comment:
Not sure I'm on board with this use of context. Just make a helper that takes the URL.